x-pack/plugins/observability/public/plugin.ts

```diff
@@ -25,15 +32,19 @@ export class Plugin implements PluginClass<ClientSetup, ClientStart> {
       appRoute: '/app/observability',
       category: DEFAULT_APP_CATEGORIES.observability,
 
-      async mount(params: AppMountParameters<unknown>) {
+      mount: async (params: AppMountParameters<unknown>) => {
         // Load application bundle
         const { renderApp } = await import('./application');
         // Get start services
         const [coreStart] = await core.getStartServices();
 
-        return renderApp(coreStart, params);
+        return renderApp(coreStart, params, this.observabilityDataAccessService.getHandlers());
```

Comment:
If we change the `getHandlers()` as suggested above it seems safe to pass on the service reference IMHO. And passing it in an object would allow for future growth without disrupting the signature too much.

```suggestion
        return renderApp(
          coreStart,
          params,
          {
            observabilityData: this.observabilityDataAccessService,
          }
        );
```

What do you think?